monai/utils/type_conversion.py

```diff
@@ -35,37 +36,32 @@
 ]
 
 
-_torch_to_np_dtype = {
-    torch.bool: np.dtype(bool),
-    torch.uint8: np.dtype(np.uint8),
-    torch.int8: np.dtype(np.int8),
-    torch.int16: np.dtype(np.int16),
-    torch.int32: np.dtype(np.int32),
-    torch.int64: np.dtype(np.int64),
-    torch.float16: np.dtype(np.float16),
-    torch.float32: np.dtype(np.float32),
-    torch.float64: np.dtype(np.float64),
-    torch.complex64: np.dtype(np.complex64),
-    torch.complex128: np.dtype(np.complex128),
-}
-_np_to_torch_dtype = {value: key for key, value in _torch_to_np_dtype.items()}
+def get_numpy_dtype_from_string(dtype: str) -> np.dtype:
+    """Get a numpy dtype (e.g., `np.float32`) from its string (e.g., `"float32"`)."""
+    return np.zeros([], dtype=dtype).dtype  # type: ignore
 
 
-def dtype_torch_to_numpy(dtype):
+def get_torch_dtype_from_string(dtype: str) -> torch.dtype:
+    """Get a torch dtype (e.g., `torch.float32`) from its string (e.g., `"float32"`)."""
+    return dtype_numpy_to_torch(get_numpy_dtype_from_string(dtype))
+
+
+def dtype_torch_to_numpy(dtype: torch.dtype) -> np.dtype:
     """Convert a torch dtype to its numpy equivalent."""
-    return look_up_option(dtype, _torch_to_np_dtype)
+    return torch.zeros([], dtype=dtype).numpy().dtype  # type: ignore
 
 
-def dtype_numpy_to_torch(dtype):
+def dtype_numpy_to_torch(dtype: np.dtype) -> torch.dtype:
     """Convert a numpy dtype to its torch equivalent."""
-    # np dtypes can be given as np.float32 and np.dtype(np.float32) so unify them
-    dtype = np.dtype(dtype) if isinstance(dtype, (type, str)) else dtype
-    return look_up_option(dtype, _np_to_torch_dtype)
+    return torch.from_numpy(np.zeros([], dtype=dtype)).dtype
```

Comment:
might be useful to cache the results because often there are just a few combinations... https://docs.python.org/3/library/functools.html#functools.cache